### PR TITLE
DDF-2985 Cleanup unused catalog-core-api dependencies

### PR DESCRIPTION
**Pull request by clockard**
#### What does this PR do?
Cleans up dependencies for the catalog core api module. This reduces false owasp findings in modules using the api. This was done on master already.
#### Who is reviewing it? 
@peterhuffer @kcover @jhunzik 

#### Ask 2 committers to review/merge the PR and tag them here.

@brjeter
@vinamartin
@brendan-hofmann 

#### How should this be tested?

Good CI Build
#### What are the relevant tickets?
For Jira: 
[DDF-2985](https://codice.atlassian.net/browse/DDF-2985)

### CODE REVIEW

**Comment by clockard** [comment]
build now

**Comment by brjeter** [comment]
Do you also plan on making a PR for downstream projects? Here's the Alliance one that went in: https://github.com/codice/alliance/commit/6dc9e989e40089ede3b6fac0eb89f0420a3708e2

**Comment by clockard** [comment]
@brjeter Yes, I was planning on doing those as well.

File: catalog/transformer/catalog-transformer-bootflag/pom.xml [hunk 1]
```diff
@@ -56,6 +56,32 @@
             <artifactId>bootflag-api</artifactId>
             <version>${project.version}</version>
         </dependency>
+        <dependency>
+            <groupId>commons-io</groupId>
```
Comment (peterhuffer) [inline catalog/transformer/catalog-transformer-bootflag/pom.xml]: ✏️ Not an issue with this PR, but this bundle should be added to the `catalog-transformer-bootflag` feature in `catalog-app`.

**Comment by clockard** [comment]
build now

**Pull request merged**